### PR TITLE
op-challenger, op-dispute-mon: Support both old and new versions of the dispute game contracts

### PR DESCRIPTION
**Pull request by ajsutton**
**Description**

Adds backwards compatibility for the fault dispute game contract original deployed to sepolia (v0.8.0).

**Tests**

Updated fault dispute game contract tests to check both versions.  Manually tested locally.

### CODE REVIEW

File: op-challenger/game/fault/contracts/faultdisputegame080.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	return time.Duration(result.GetUint64(0)) * time.Second / 2, nil
+}
+
+func (f *FaultDisputeGameContract080) GetClaim(ctx context.Context, idx uint64) (types.Claim, error) {
+	claim, err := f.FaultDisputeGameContractLatest.GetClaim(ctx, idx)
+	if err != nil {
+		return types.Claim{}, err
+	}
+	// Replace the resolved sentinel with what the bond would have been
+	if claim.Bond.Cmp(resolvedBondAmount) == 0 {
```
Comment (Inphi) [inline op-challenger/game/fault/contracts/faultdisputegame080.go:72]: The old sentinel value is uint128 max - 
https://github.com/ethereum-optimism/optimism/blob/842ed1dd927b35febd1c803500b647971f89992c/packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol#L61

Note that op-dispute-mon had a similar [bug](https://github.com/ethereum-optimism/optimism/blob/8c3a849de99d28e35d93f6e81e0074a24c3fcf60/op-dispute-mon/mon/bonds/collateral.go#L26) before we stopped using the sentinel value.

File: op-challenger/game/fault/contracts/faultdisputegame080.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+func (f *FaultDisputeGameContract080) GetAllClaims(ctx context.Context, block rpcblock.Block) ([]types.Claim, error) {
+	claims, err := f.FaultDisputeGameContractLatest.GetAllClaims(ctx, block)
+	if err != nil {
+		return nil, err
+	}
+	resolvedClaims := make([]*types.Claim, 0, len(claims))
+	positions := make([]*big.Int, 0, len(claims))
+	for i, claim := range claims {
+		if claim.Bond.Cmp(resolvedBondAmount) == 0 {
```
Comment (Inphi) [inline op-challenger/game/fault/contracts/faultdisputegame080.go:90]: Sentinel check should be for uint128 max

File: op-challenger/game/fault/contracts/faultdisputegame080.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+func (f *FaultDisputeGameContract080) IsResolved(ctx context.Context, block rpcblock.Block, claims ...types.Claim) ([]bool, error) {
+	rawClaims, err := f.FaultDisputeGameContractLatest.GetAllClaims(ctx, block)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get raw claim data: %w", err)
+	}
+	results := make([]bool, len(claims))
+	for i, claim := range claims {
+		results[i] = rawClaims[claim.ContractIndex].Bond.Cmp(resolvedBondAmount) == 0
```
Comment (Inphi) [inline op-challenger/game/fault/contracts/faultdisputegame080.go:112]: Update the bond sentinel value.

**Review by Inphi — COMMENTED** [review]
The sentinel value used to detect resolved claims needs to be fixed. Other than that, everything else looks good.

Having op-dispute-mon rely on the `FaultDisputeGame` interface for enrichment helped simplify this change in hindsight.

File: op-challenger/game/fault/contracts/faultdisputegame080.go (hunk 1 above)
Comment (Inphi) [reply op-challenger/game/fault/contracts/faultdisputegame080.go:72]: This is false. No issues here.

**Review by Inphi — APPROVED** [review]

**Review by refcell — APPROVED** [review]

**Pull request merged**